### ISSUE TITLE
False positive: method has incorrect number of parameters for trait

### ISSUE THREAD

**Issue by Person-93** [intellij-rust/intellij-rust#3239]
## Environment

* **Intellij-Rust plugin version:** v0.2.0.2113-183
* **Rust toolchain version:** rustc 1.31.1 (b6c32da9b 2018-12-18)
* **IDE name and version:** CLion 2018.3.2
* **Operating system:** Debian Buster

## Problem description
If a trait has two versions of a method controlled by a `#[cfg(...)]` statement, the implementation will always be flagged as an error and sometimes the second declaration will be as well.

## Steps to reproduce
Here is a simple example:
```rust
trait ConfigurableTrait {
    #[cfg(no_count)]
    fn do_stuff(&self);
    #[cfg(not(no_count))]
    fn do_stuff(&self, count: u8);
}

struct Thing;

impl ConfigurableTrait for Thing {
    fn do_stuff(&self, _: u8) {}
}

fn main() {
    println!("Hello, world!");
}

**Comment by ortem** [comment]
Looks like it was fixed in #4537. @Person-93 is this works fine now?
